pkg/provider/aws/resources/vpc.go

```diff
@@ -133,7 +135,7 @@ func CreateInterfaceVpcEndpoint(service string, vpc *Vpc, region *Region, dag *c
 
 func (vpc *Vpc) GetVpcSubnets(dag *core.ResourceGraph) []*Subnet {
 	subnets := []*Subnet{}
-	downstreamDeps := dag.GetDownstreamResources(vpc)
+	downstreamDeps := dag.GetUpstreamResources(vpc)
```

Comment:
This is the bug fix, the rest is updating the unit tests to have caught this bug.